File: sdk/typescript/src/identities.ts
/**
 * TypeScript SDK client for Phase 5.48 Identity Assurance platform.
 */

export interface IdentityReference {
  identity_id: string;
  tenant_id: string;
  name: string;
  identity_type: string;
  category: string;
  status: string;
  external_id?: string;
  created_at: string;
}

export interface IdentityTrustAssessment {
  assessment_id: string;
  identity_id: string;
  tenant_id: string;
  is_trusted: boolean;
  risk_level: string;
  evaluated_at: string;
}

export class IdentityAssuranceClient {
  constructor(private baseUrl: string, private apiKey?: string) {}

  private async request(path: string, options: RequestInit = {}, tenantId = "default_tenant"): Promise<any> {
    const headers: Record<string, string> = {
      "Content-Type": "application/json",
      "x-tenant-id": tenantId,
      ...(this.apiKey ? { Authorization: `Bearer ${this.apiKey}` } : {}),
      ...(options.headers as Record<string, string>),
    };

    const res = await fetch(`${this.baseUrl}${path}`, { ...options, headers });
    if (!res.ok) {
      throw new Error(`Identity Assurance API error: ${res.statusText}`);
    }
    return res.json();
  }

  async registerIdentity(
    name: string,
    identityType = "HUMAN",
    category = "EMPLOYEE",
    externalId?: string,
    tenantId = "default_tenant"
  ): Promise<IdentityReference> {
    return this.request(
      "/v1/identities",
      {
        method: "POST",
        body: JSON.stringify({ name, identity_type: identityType, category, external_id: externalId }),
      },
      tenantId
    );
  }

  async getIdentity(identityId: string, tenantId = "default_tenant"): Promise<IdentityReference> {
    return this.request(`/v1/identities/${identityId}`, { method: "GET" }, tenantId);
  }

  async assessTrust(identityId: string, tenantId = "default_tenant"): Promise<IdentityTrustAssessment> {
    return this.request(`/v1/identities/${identityId}/trust`, { method: "POST" }, tenantId);
  }

  async evaluateAssurance(identityId: string, tenantId = "default_tenant"): Promise<any> {
    return this.request(`/v1/identities/${identityId}/assurance`, { method: "GET" }, tenantId);
  }
}

File: sdk/go/identities.go
package llmengine

import (
	"bytes"
	"encoding/json"
	"fmt"
	"net/http"
)

type IdentityReference struct {
	IdentityID   string `json:"identity_id"`
	TenantID     string `json:"tenant_id"`
	Name         string `json:"name"`
	IdentityType string `json:"identity_type"`
	Category     string `json:"category"`
	Status       string `json:"status"`
	ExternalID   string `json:"external_id,omitempty"`
}

type IdentityTrustAssessment struct {
	AssessmentID string  `json:"assessment_id"`
	IdentityID   string  `json:"identity_id"`
	TenantID     string  `json:"tenant_id"`
	IsTrusted    bool    `json:"is_trusted"`
	RiskLevel    string  `json:"risk_level"`
}

type IdentityAssuranceService struct {
	client *Client
}

func (s *IdentityAssuranceService) RegisterIdentity(tenantID, name, identityType, category string) (*IdentityReference, error) {
	payload := map[string]string{
		"name":          name,
		"identity_type": identityType,
		"category":      category,
	}
	body, _ := json.Marshal(payload)

	req, err := http.NewRequest("POST", fmt.Sprintf("%s/v1/identities", s.client.BaseURL), bytes.NewBuffer(body))
	if err != nil {
		return nil, err
	}
	req.Header.Set("Content-Type", "application/json")
	req.Header.Set("x-tenant-id", tenantID)

	resp, err := s.client.HTTPClient.Do(req)
	if err != nil {
		return nil, err
	}
	defer resp.Body.Close()

	var identity IdentityReference
	if err := json.NewDecoder(resp.Body).Decode(&identity); err != nil {
		return nil, err
	}
	return &identity, nil
}
